### ISSUE TITLE
`use_build_context_synchronously` triggered on `switch` with `break`

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#4741]
**Describe the issue**
`use_build_context_synchronously` triggered on `switch` with `break`

**To Reproduce**
```dart
void main() async {
  int i = 0;
  late BuildContext context;
  switch (i) {
    case 0:
      await f(context);
      break;
    case 1:
      await f(context); //lint
      break;
  }
}

Future<void> f(BuildContext context) async {}
```

**Expected behavior**
Since I'm using break, that shouldn't trigger the lint.

**Comment by srawlins** [comment]
Duplicate of https://github.com/dart-lang/linter/issues/4705
